src/lookup/derive.ts

```diff
@@ -23,13 +23,68 @@ import {
   concatenateContexts,
   defaultContext,
   Iri,
-  isVerifiablePresentation,
   VerifiableCredential,
 } from "../common/common";
 import fallbackFetch from "../fetcher";
+import { query, VerifiablePresentationRequest } from "./query";
 
 const INCLUDE_EXPIRED_VC_OPTION = "ExpiredVerifiableCredential" as const;
 
+/**
+ * This creates the proprietary data structure used for querying the legacy /derive
+ * endpoint of ESS.
+ *
+ * @param vcShape The VC example used for the VP request
+ * @param includeExpiredVc An option to query for expired VC as well
+ * @returns A legacy object expected by the /derive endpoint of the ESS 2.0 VC service
+ */
+function buildLegacyQuery(
+  vcShape: Partial<VerifiableCredential>,
+  includeExpiredVc: boolean
+) {
+  // credentialClaims should contain all the claims, but not the context.
+  const { "@context": claimsContext, ...credentialClaims } = vcShape;
+  const credentialRequestBody: {
+    verifiableCredential: Partial<VerifiableCredential>;
+    options?: { include: typeof INCLUDE_EXPIRED_VC_OPTION };
+  } = {
+    // See https://w3c-ccg.github.io/vc-api/holder.html
+    verifiableCredential: {
+      "@context": concatenateContexts(defaultContext, claimsContext),
+      ...credentialClaims,
+    },
+  };
+  if (includeExpiredVc) {
+    credentialRequestBody.options = {
+      include: INCLUDE_EXPIRED_VC_OPTION,
+    };
+  }
+
+  return credentialRequestBody;
+}
+
+/**
+ * See https://w3c-ccg.github.io/vp-request-spec/#query-by-example.
+ * @param vcShape The VC example used for the VP request
+ * @returns A Query by Example VP Request based on the provided example.
+ */
+function buildQueryByExample(
```

Comment:
This feels like it's only used for mocks / testing? Not a public API that's useful, may I suggest moving it to a `mocks` or `testing` folder/module?

Reply:
Not quite: it is used to transform the input of `getVerifiableCredentialAllFromShape` into a valid VPR. It isn't a public API because it's only used internally by `getVerifiableCredentialAllFromShape`, but it is not test-only code.

Reply:
I think this might just be a gap in my knowledge around VC/VPRs